pkg/kubectl/cmd/delete.go

```diff
@@ -248,13 +261,20 @@ func ReapResult(r *resource.Result, f cmdutil.Factory, out io.Writer, isDefaultD
 		if err != nil {
 			return err
 		}
+		// Set WaitForDeletion to false for namespace deletion unless it was explicitly set by the user.
+		if info.Mapping.Resource == "Namespace" && isDefaultWaitForDeletion {
```

Comment:
This is required to not break existing functionality. Kubectl does not wait for namespace deletion, while it waits for all others (example: deployment, replicasets).

Updated the check to use "namespaces" and also added a test for this.